code/processes/gateway.q

```diff
@@ -483,7 +483,7 @@ asyncexec:asyncexecjpt[;;raze;();0Wn]
 // execute a synchronous query
 syncexecjpre36:{[query;servertype;joinfunction]
  // Check correct function called
- if[not .gw.call .z.w;
+ if[(.z.w<>0)&(.z.w in key .gw.call)&not .gw.call .z.w;
```

Comment:
do you need both these checks? will `0` be in `key .gw.call`? my understanding is it wouldn't be and if that's the case, I think the second condition implies the first?

Reply:
Zero is in .gw.call after the first call. The second one is to allow for calls to work in qcon